ansible/roles/db/tasks/main.yml

```diff
@@ -0,0 +1,47 @@
+---
+# tasks file for roles/db
+- name: Install base packages
+  apt:
+    name: "{{ prepare_host_packages }}"
+    state: present
+    update_cache: true
+  loop: "{{ prepare_host_packages }}"
+
+- name: Install pymysql python package
+  ansible.builtin.pip:
+    name: pymysql
+    state: present
+
+- name: Create MySQL user
+  community.mysql.mysql_user:
+    name: "{{ mysql_user }}"
+    password: "{{ mysql_password }}"
+    host: "%"
+    priv: '*.*:ALL'
+    login_unix_socket: /var/run/mysqld/mysqld.sock
+    login_user: root
+    state: present
+
+- name: Create flask_db database
+  community.mysql.mysql_db:
+    name: "{{ mysql_db }}"
+    state: present
+    login_unix_socket: /var/run/mysqld/mysqld.sock
+    login_user: root
+
+- name: Grant privileges on flask_db to admin
+  community.mysql.mysql_user:
```

Comment:
Module `community.mysql.mysql_user`  used twice.
You can just create database first and after that create user with required privileges.